scripts/xtensa-build-zephyr.py

```diff
@@ -1169,6 +1175,29 @@ def reproducible_checksum(platform, ri_file):
 	chk256 = sof_ri_info.EraseVariables(ri_file, parsed_ri, west_top / repro_output)
 	print('sha256sum {0}\n{1} {0}'.format(repro_output, chk256))
 
+def create_tarball():
+	# vendor directories (typically under /lib/firmware) to include
+        # in the tarball
+	vendor_dirs = [ 'intel']
+
+	build_top = west_top / 'build-sof-staging' / 'sof'
+
+	tarball_name = 'sof.tgz'
+
+	prev_cwd = os.getcwd()
+	os.chdir(build_top)
+
+	tarball_path = build_top / tarball_name
+	with tarfile.open(tarball_path, 'w:gz') as tarball:
+		for vendor_dir in vendor_dirs:
+			vendor_dir_path = build_top / vendor_dir
+			if not vendor_dir_path.exists():
+				continue
+			tarball.add(vendor_dir, recursive=True)
+		print(f"Tarball %s created" % tarball_path)
+		tarball.close()
```

Comment:
I think `with` does that for you but no big deal.

Reply:
correct, with will close it on context exit

Reply:
Dropped the close() and let with handle it.